beacon_node/lighthouse_network/src/peer_manager/network_behaviour.rs

```diff
@@ -156,8 +156,10 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
             BanResult::BadScore => {
                 // This is a faulty state
                 error!(self.log, "Connected to a banned peer. Re-banning"; "peer_id" => %peer_id);
-                // Reban the peer
+                // Disconnect the peer.
                 self.goodbye_peer(&peer_id, GoodbyeReason::Banned, ReportSource::PeerManager);
+                // Re-ban the peer to prevent repeated errors.
+                self.events.push(PeerManagerEvent::Banned(peer_id, vec![]));
```

Comment:
shouldn't the prev line handle this? I tracked it down some calls and I think it should but if it for some reason doesn't we get out of sync with the swarm banned peers. I don't actually have a better suggestion here tho

Reply:
Yeah, so I know this is super convoluted. We need to re-write this whole thing at some point. 

The goodbye peer thing, called report_peer() in peerdb, which gives a score transition. If it transitions to a banned state, then we ban it. But if its already banned, we do nothing. 
See here
https://github.com/sigp/lighthouse/blob/stable/beacon_node/lighthouse_network/src/peer_manager/peerdb.rs#L535
and then here:
https://github.com/sigp/lighthouse/blob/stable/beacon_node/lighthouse_network/src/peer_manager/peerdb.rs#L1116

So the problem is that we are already out of sync with the swarm. If peers are connecting to a banned peer, it means that the swarm thinks its not banned but our state thinks it is banned. 

The report_peer() function won't tell the swarm to ban again, so I manually intervened. In principle we shouldn't hit this code, this is more of a fail-over if we go out of sync for some reason.

Reply:
I dont dare have the sense that I totally grasp all aspects of this issue.. that said, might we consider some time based deferral for a banned peer.. 

eg.. if a bad peer tries one, it is locked out for 10min (example).. and if it tried again within 10min, then it is extended to an hour.. and then a day.. and then a week.. etc.. something such that there is a time-based memory to ensure bad peers are just flat ignored for a while.. otherwise I would think there is some aspect of a ddos possible

Reply:
We already do this in the general case. However here we are dealing with an out of sync db. Let's merge this in the meantime since a proper solution is a massive task for which we don't have the bandwidth atm